src/Event/Event.js

```diff
@@ -1,38 +1,150 @@
+      this.isPressDisabled() &&
+      this.isDragEnabled() &&
+      hasMovedEnough(gestureState),
+    onPanResponderMove: Animated.event(
+      [
+        null,
         {
-          backgroundColor: event.color,
+          dx: this.translatedByDrag.x,
+          dy: this.translatedByDrag.y,
         },
-        containerStyle,
-      ]}
-      disabled={!onPress && !onLongPress}
-    >
-      {EventComponent ? (
-        <EventComponent event={event} position={position} />
-      ) : (
-        <Text style={styles.description}>{event.description}</Text>
-      )}
-    </TouchableOpacity>
-  );
-};
+      ],
+      {
+        useNativeDriver: false,
+      },
+    ),
+    onPanResponderTerminationRequest: () => false,
+    onPanResponderRelease: (_, gestureState) => {
+      const { dx, dy } = gestureState;
+      this.onDragRelease(dx, dy);
+    },
+    onPanResponderTerminate: () => {
+      this.translatedByDrag.setValue({ x: 0, y: 0 });
+    },
+  });
+
+  constructor(props) {
+    super(props);
+
+    const { left, width } = props.position;
+    this.currentWidth = new Animated.Value(width);
+    this.currentLeft = new Animated.Value(left);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.position !== this.props.position) {
+      this.translatedByDrag.setValue({ x: 0, y: 0 });
+      const animations = [];
+      const { left, width } = this.props.position;
+      if (prevProps.position.width !== width) {
+        animations.push(
+          Animated.timing(this.currentWidth, {
+            toValue: width,
+            duration: UPDATE_EVENT_ANIMATION_DURATION,
+            useNativeDriver: false,
+          }),
+        );
+      }
+      if (prevProps.position.left !== left) {
+        animations.push(
+          Animated.timing(this.currentLeft, {
+            toValue: left,
+            duration: UPDATE_EVENT_ANIMATION_DURATION,
+            useNativeDriver: false,
+          }),
+        );
+      }
+      Animated.parallel(animations).start();
+    }
+  }
+
+  isPressDisabled = () => {
+    return !this.props.onPress && !this.props.onLongPress;
+  };
+
+  isDragEnabled = () => {
+    return !!this.props.onDrag;
+  };
+
+  onDragRelease = (dx, dy) => {
+    const { position, onDrag } = this.props;
+    if (!onDrag) {
+      return;
+    }
+
+    const newX = position.left + position.width / 2 + dx;
+    const newY = position.top + dy;
+    onDrag(this.props.event, newX, newY);
+  };
+
+  render() {
+    const {
+      event,
+      onPress,
+      onLongPress,
+      position,
+      EventComponent,
+      containerStyle,
+    } = this.props;
+
+    const { top, height } = position;
+    return (
+      <Animated.View
+        style={[
+          styles.container,
+          {
+            top,
+            left: this.currentLeft,
+            height,
+            width: this.currentWidth,
+            backgroundColor: event.color,
+          },
+          this.translatedByDrag.getTranslateTransform(),
```

Comment:
From the doc, it seems we have to map with `transform` property, https://reactnative.dev/docs/animatedvaluexy#gettranslatetransform. Is it still working?

Reply:
You are right. For some reason I can't remember, I did it like this (some time ago), and it works. I'm checking into it :thinking: